File: uicoremk/src/main/java/com/mozhimen/uicoremk/refreshmk/TextOverView.java
package com.mozhimen.uicoremk.refreshmk;

import android.content.Context;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;
import android.view.animation.LinearInterpolator;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.mozhimen.uicoremk.R;
import com.mozhimen.uicoremk.refreshmk.commons.RefreshMKOverView;

public class TextOverView extends RefreshMKOverView {
    private TextView mText;
    private View mRotateView;

    public TextOverView(@NonNull Context context) {
        super(context);
    }

    public TextOverView(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
    }

    public TextOverView(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void init() {
        LayoutInflater.from(getContext()).inflate(R.layout.refreshmk_overview_text, this, true);
        mText = findViewById(R.id.refreshmk_overview_text_content);
        mRotateView = findViewById(R.id.refreshmk_overview_text_icon);
    }

    @Override
    public void onScroll(int scrollY, int pullRefreshHeight) {
    }

    @Override
    public void onVisible() {
        mText.setText("下拉刷新");
    }

    @Override
    public void onOver() {
        mText.setText("松开刷新");
    }

    @Override
    public void onRefresh() {
        mText.setText("正在刷新...");
        Animation operatingAnim = AnimationUtils.loadAnimation(getContext(), R.anim.refreshmk_rotate);
        LinearInterpolator interpolator = new LinearInterpolator();
        operatingAnim.setInterpolator(interpolator);
        mRotateView.setAnimation(operatingAnim);
    }

    @Override
    public void onFinish() {
        mRotateView.clearAnimation();
    }
}

File: uicorek/src/main/java/com/mozhimen/uicorek/layoutk/refresh/temps/TextOverView.kt
package com.mozhimen.uicorek.layoutk.refresh.temps

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.TextView
import com.mozhimen.basick.extsk.startRotate
import com.mozhimen.basick.extsk.stopAnim
import com.mozhimen.basick.utilk.UtilKRes
import com.mozhimen.uicorek.R
import com.mozhimen.uicorek.layoutk.refresh.commons.RefreshOverView

/**
 * @ClassName TextOverView
 * @Description TODO
 * @Author mozhimen / Kolin Zhao
 * @Date 2022/4/17 23:52
 * @Version 1.0
 */
class TextOverView @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
    RefreshOverView(context, attrs, defStyleAttr) {
    private lateinit var _titleView: TextView
    private lateinit var _animView: View

    override fun init() {
        LayoutInflater.from(context).inflate(R.layout.layoutk_refresh_overview_text, this, true)
        _titleView = findViewById(R.id.layoutk_refresh_overview_text_title)
        _animView = findViewById(R.id.layoutk_refresh_overview_text_img)
    }

    override fun onScroll(scrollY: Int, pullRefreshHeight: Int) {}

    override fun onVisible() {
        _titleView.text = UtilKRes.getString(R.string.layoutk_refresh_visible)
    }

    override fun onOverflow() {
        _titleView.text = UtilKRes.getString(R.string.layoutk_refresh_overflow)
    }

    override fun onStartRefresh() {
        _titleView.text = UtilKRes.getString(R.string.layoutk_refresh_refreshing)
        _animView.startRotate()
    }

    override fun onFinish() {
        _animView.stopAnim()
    }
}